opencog/atoms/bind/PatternLink.cc

```diff
@@ -639,6 +643,35 @@ void PatternLink::make_map_recursive(const Handle& root, const Handle& h)
 	}
 }
 
+// Hack alert: Definitely it should not be here. Though some refactoring
+// regarding shared libraries circular dependencies (liblambda and libquery)
```

Comment:
Ugh. the circular dependencies are killing me. If you have any good ideas, let me know.